docs/docs/code/bash/README.md

```diff
@@ -0,0 +1,199 @@
+
+## Logging & debugging
+
+When it comes to debugging bash scripts, `echo` is your friend.
+
+Your `echo` statements will print their output in the workflow step results:
+
+```bash
+MESSAGE='Hello world'
+
+# The message will now be available in the "Result > Logs" area in the workflow step
+echo $MESSAGE
+```
+
+## Available binaries
+
+Bash steps come with many common & useful binaries preinstalled and available in `$PATH` for you to use out of the box. These binaries include but aren't limited to:
+
+* `curl` for making HTTP requests
+* `jq` for manipulating and viewing JSON data
+* `git` for interacting with remote repositories
+
+Unfortunately it is not possible to install from a package manager like `apt` or `yum`.
+
+If you need a package pre-installed in your bash steps, [just ask us](https://pipedream.com/support).
+
+Otherwise, you can use `/tmp` to download and install from source.
+
+## Making an HTTP request
+
+`curl` is already preinstalled in bash steps, we recommend using it for making HTTP requests in your code for sending or requesting data from APIs or webpages.
+
+### Making a GET request
+
+You can use `curl` to perform GET requests from websites or APIs directly.
+
+```bash
+# Get the current weather in San Francisco
+WEATHER=`curl --silent https://wttr.in/San\ Francisco\?format=3`
+
+echo $WEATHER
+# Produces:
+# San Francisco: 🌫  +48°F
+```
+
+::: tip
+Use the `--silent` flag with `curl` to suppress extra extra diagnostic information that `curl` produces when making requests.
+
+This enables you to only worry about the body of the response so you can visualize it with tools like `echo` or `jq`. 
+:::
+
+### Making a POST request
+
+`curl` handles making POSTs requests as well. The `-X` flag allow you to specify the HTTP method you'd like to use for an HTTP request.
+
+The `-d` flag is for passing data in the POST request.
+
+```bash
+curl --silent -X POST https://postman-echo.com/post -d 'name=Bulbasaur&id=1'
+
+# to store the API response in a variable, interpolate the response into a string and store it in variable
+RESPONSE=`curl --silent -X POST https://postman-echo.com/post -d 'name=Bulbasaur&id=1'`
+
+# now the response is stored as a variable
+echo $RESPONSE
+```
+
+### Using API key authentication
+
+Some APIs require you to use a secret API key to prove your access.
+
+`curl` has an `-h` flag where you can pass your API key as a token.
+
+Below is an example of searching Twitter for mentions of a specific Twitter handle. Twitter requires a valid API key passed in the header of the request.
+
+```bash
+curl --silent -X POST -h "authorization:Bearer $(<your api key here>)" https://api.twitter.com/2/users/@pipedream/mentions
+```
+
+## Sharing data between steps
+
+A step can accept data from other steps in the same workflow, or pass data downstream to others.
+
+This makes your steps even more powerful, you can compose new workflows and reuse steps.
```

Comment:
Consider removing this sentence, since this section isn't about composition. Then you can move straight from: "A step can receive data from other steps and pass it to others" -> here's how to use data from other steps.